src/Service/ImageHandler.php

```
@@ -119,4 +137,22 @@ private function hash(string $url): string
 
         return hash('sha256', $parts['host'] ?? 'unknown');
     }
+
+    /**
+     * Create local filename based on URL and mine-type.
+     *
+     * @param string $url
+     *   URL for the file to generate name for
+     * @param string $mimetype
+     *   The files mine-type
+     *
+     * @return string
+     *   Generated local filename
+     */
+    private function generateLocalFilename(string $url, string $mimetype): string
+    {
+        $ext = (new MimeTypes())->getExtensions($mimetype)[0];
```

This will fail hard if the `MimeTypes` doesn't know the passed mime type.